src/modules/ekf2/EKF/aid_sources/range_finder/range_finder_consistency_check.hpp

```diff
@@ -74,6 +77,12 @@ class RangeFinderConsistencyCheck final
 	uint64_t _time_last_inconsistent_us{};
 	uint64_t _time_last_horizontal_motion{};
 
+	bool _is_blocked{false};
+	float _max_fog_dist{0.f};
+	math::MedianFilter<float, 11> _distance_filter{};
```

Comment:
maybe a window of 3 or 5 might be enough if we expect to only have single outliers?